dnf/module/repo_module_version.py

```diff
@@ -0,0 +1,155 @@
+    def __repr__(self):
+        return self.full_version
+
+    def report_profile_error(self, profile, default_profiles_used=False):
+        if default_profiles_used:
+            msg = module_errors[NO_PROFILE_SPECIFIED].format(self.name)
+        else:
+            msg = module_errors[NO_PROFILE_ERR].format(profile)
+            if self.profiles:
+                msg += " " + module_errors[POSSIBLE_PROFILES].format(self.profiles)
+            else:
+                msg += " " + module_errors[NO_PROFILES_AVAILABLE]
+
+        raise Error(msg)
+
+    def install(self, profiles, default_profiles):
+        self._install_profiles(profiles, False)
+        self._install_profiles(default_profiles, True)
+
+        profiles.extend(self.repo_module.conf.profiles)
+        profiles.extend(default_profiles)
+        self.base._module_persistor.set_data(self.repo_module, stream=self.stream,
+                                             version=self.version, profiles=sorted(set(profiles)))
+
+    def _install_profiles(self, profiles, defaults_used):
+        for profile in profiles:
+            if profile not in self.profiles:
+                self.report_profile_error(profile, defaults_used)
+
+            for single_nevra in self.profile_nevra(profile):
+                subject = Subject(single_nevra)
+                nevra_obj = list(subject.get_nevra_possibilities(hawkey.FORM_NEVR))[0]
+
+                self.base.install(single_nevra, reponame=self.repo.id, forms=hawkey.FORM_NEVR)
+                self.base._goal.group_members.add(nevra_obj.name)
+
+    def upgrade(self, profiles):
+        for profile in profiles:
+            if profile not in self.profiles:
+                self.report_profile_error(profile)
+
+            for single_nevra in self.profile_nevra(profile):
+                self.base.upgrade(single_nevra, reponame=self.repo.id)
+
+        self.base._module_persistor.set_data(self.repo_module, stream=self.stream,
+                                             version=self.version)
+
+    def remove(self, profiles):
+        for profile in profiles:
+            if profile not in self.profiles:
+                self.report_profile_error(profile)
+
+            for single_nevra in self.profile_nevra(profile):
+                remove_query = dnf.subject.Subject(single_nevra) \
+                    .get_best_query(self.base.sack, forms=hawkey.FORM_NEVR)
+
+                if self.base._yumdb.get_package(remove_query[0]).reason == 'user':
+                    continue
+
+                self.base._remove_if_unneeded(remove_query)
+
+        conf = self.repo_module.conf
+        version = conf.version
+        profiles = [x for x in conf.profiles if x not in profiles]
+
+        if len(conf.profiles) == 0:
+            conf.version = -1
+
+        self.base._module_persistor.set_data(self.repo_module, stream=self.stream, version=version,
+                                             profiles=sorted(set(profiles)))
+
+    def nevra(self):
```

Comment:
Is that really good idea?
Solution we should fix nevra filter
or we should drop nevra filter.